WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt

```diff
@@ -1522,6 +1517,65 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
             refreshIconCounts()
             initBookmarkButton()
         }
+
+        private fun createPostDetailsHeaderUiState(
+            post: ReaderPost,
+            headerView: ReaderPostDetailHeaderView
+        ): ReaderPostDetailsHeaderUiState {
+            val onBlogSectionClicked = { postId: Long?, blogId: Long? ->
+                ReaderActivityLauncher.showReaderBlogPreview(context, post)
+            }
+            val onFollowButtonClicked = { toggleFollowStatus(headerView.header_follow_button) }
+            return postDetailsHeaderViewUiStateBuilder.mapPostToUiState(
+                    post,
+                    onBlogSectionClicked,
+                    onFollowButtonClicked
+            )
+        }
+    }
+
+    // TODO: Refactor it
+    private fun toggleFollowStatus(followButton: ReaderFollowButton) {
+        if (!NetworkUtils.checkConnection(context)) {
+            return
+        }
+        post?.let { post ->
+            val isAskingToFollow = !post.isFollowedByCurrentUser
+            if (isAskingToFollow) {
+                onFollowTapped(followButton, post.blogName, post.blogId)
+            } else {
+                onFollowingTapped()
+            }
+
+            val listener = ReaderActions.ActionListener { succeeded ->
+                if (context == null) {
+                    return@ActionListener
+                }
+                followButton.isEnabled = true
+                if (succeeded) {
+                    post.isFollowedByCurrentUser = isAskingToFollow
+                } else {
+                    val errResId = if (isAskingToFollow) {
+                        string.reader_toast_err_follow_blog
```

Comment:
NitPick: Lint sometime fails because it marks resources which are missing the "R." prefix as unused. We might want to consider adding the prefix and removing the import. Wdyt?

Reply:
Fixed in c6acf92